scipy/stats/_qmc.py

```diff
@@ -976,32 +1004,52 @@ class LatinHypercube(QMCEngine):
     >>> qmc.discrepancy(sample)
     0.0176...  # random
 
+    Use the `orthogonal` keyword argument to produce an orthogonal array based
+    LHS. The construction imposes to use a square of a prime number.
+
+    >>> sampler = qmc.LatinHypercube(d=2, orthogonal=True)
+    >>> sample = sampler.random(n=9)
+    >>> qmc.discrepancy(sample)
+    0.00526...  # random
+
+    Options could be combined as a same time to produce an optimized centered
+    orthogonal array based LHS. After optimization, the result would not
+    be guaranteed to be orthogonal.
+
     """
 
     def __init__(
         self, d: IntNumber, *, centered: bool = False,
+        orthogonal: bool = False,
         optimization: Optional[Literal["random-cd"]] = None,
         seed: SeedType = None
     ) -> None:
         super().__init__(d=d, seed=seed)
         self.centered = centered
 
-        lhs_methods: Dict[Optional[Literal["random-cd"]], Callable] = {
-            None: self._random,
+        optimization_method: Dict[Literal["random-cd"], Callable] = {
             "random-cd": self._random_cd,
         }
 
-        try:
-            if optimization is not None:
+        self.optimization_method: Optional[Callable]
+        if optimization is not None:
+            try:
                 optimization = optimization.lower()  # type: ignore[assignment]
-            self.lhs_method = lhs_methods[optimization]
-        except KeyError:
-            raise ValueError(f"{optimization!r} is not a valid optimization"
-                             " method. It must be one of"
-                             f" {set(lhs_methods)!r}")
+                self.optimization_method = optimization_method[optimization]
+            except KeyError:
+                raise ValueError(f"{optimization!r} is not a valid"
+                                 " optimization method. It must be one of"
+                                 f" {set(optimization_method)!r}")
+
+            self._n_nochange = 100
+            self._n_iters = 10_000
+        else:
+            self.optimization_method = None
 
-        self._n_nochange = 100
-        self._n_iters = 10_000
+        if not orthogonal:
+            self.lhs_method = self._random
+        else:
+            self.lhs_method = self._random_oa_lhs
```

Comment:
If we use the dictionary approach to select the optimization method, why don't we use the dictionary approach to select the generation method?

Reply:
Since I am going with the other approach with optimize I can move this back in the dictionnary. Closing this discussion in favour of the other.

Reply:
Actually here I found it simpler to use this if as we just have a boolean. But if we would change the parameter to accept an int 1 or 2, then it might be better to have the dictionary. Is it ok with you? Or I am missing something?

Reply:
If we change the parameter to accept 1 or 2, then I agree with changing it to a dictionary.
If we keep the parameter accepting a boolean, then it is OK with me to change it, and it is OK with me to leave it as-is if you have a reason for it. 
Assuming this was to remain a boolean, I was questioning why not use the dictionary approach since the dictionary approach is already used above. I agree that it is more natural to use an "if ... else" approach for a boolean variable than it is to use an "if .. else" approach for a string variable. But I would tend toward the dictionary approach, even for booleans, if the dictionary approach is already used elsewhere in the same function, simply as a matter of consistency of style within a given function.

Reply:
OK I will give this a try.